src/units/attack_type.hpp

```diff
@@ -82,8 +82,31 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	 */
 	bool has_special(const std::string& special, bool simple_check = false) const;
 	unit_ability_list get_specials(const std::string& special) const;
-	std::vector<std::pair<t_string, t_string>> special_tooltips(boost::dynamic_bitset<>* active_list = nullptr) const;
-	std::vector<std::pair<t_string, t_string>> abilities_special_tooltips(boost::dynamic_bitset<>* active_list = nullptr) const;
+
+	/**
+	 * Return value for special_tooltips() and abilities_special_tooltips().
+	 *
+	 * Those functions are responsible for choosing whether to use
+	 * name_inactive, name_affected, etc.
+	 */
+	struct attack_tooltip_metadata
+	{
+		// constructor because emplace_back() doesn't use aggregate initialization
```

Comment:
Don’t add a ctor. You can use the `AGGREGATE_EMPLACE` macro in place of `emplace_back`.